test/python/qpy/test_block_load_from_qpy.py

```diff
@@ -187,6 +194,20 @@ def test_nested_blocks(self):
                     builder.delay(200, DriveChannel(1))
         self.assert_roundtrip_equal(test_sched)
 
+    def test_called_schedule(self):
```

Comment:
One thing I think would be good is to maybe have a qpy test with the rzx calibration passes. Since a big part of this PR is fixing the QPY support with the pass output having a test to verify the pass output can be round tripped through qpy is simple way to ensure we don't regress on this moving forward.

Reply:
Thanks Matthew. I added new tests for QPY. I also refactored the RZX builder test in [8562d98](https://github.com/Qiskit/qiskit-terra/pull/8949/commits/8562d98393b5c8a55b49f3d601f04314f261c17c) since the output schedule has been never really tested. This is why the alignment error has been existed before this PR (this PR unintentionally fixed it).